packages/compiler/src/core/numeric.ts

```diff
@@ -118,14 +118,16 @@ function parse(original: string): InternalData {
     }
 
     let end = stringValue.length;
-    while (stringValue[end - 1] === "0") {
-      if (end < adjustedPointIndex) {
-        // if we are looking at a zero before the decimal point, we need to decrease the exponent
-        exp++;
-      } else {
+    let index = end;
+    while (stringValue[index - 1] === "0") {
```

Comment:
I think this simplified version does the job:

```typescript
   let index = stringValue.length;
   while (stringValue[index - 1] === "0" && index > adjustedPointIndex) {
      index--;
}
```

Reply:
true